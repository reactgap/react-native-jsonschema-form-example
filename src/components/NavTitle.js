// @flow
// @format

import React, { PureComponent } from 'react'
import { StyleSheet, Text, type TextStyle } from 'react-native'
import csstyles from '../csstyles'
import { IS_DEVICE_VERY_SHORT_WIDTH } from '../utils/deviceHelper'

type Props = {
  title?: string,
  style?: TextStyle,
  numberOfLines?: number
}

class NavTitle extends PureComponent<Props> {
  render() {
    const { titleCode, t, style, title, numberOfLines } = this.props

    return (
      <Text style={[styles.title, style]} numberOfLines={numberOfLines}>
        {titleCode ? titleCode : title}
      </Text>
    )
  }
}

const navTitleLeftPadding = IS_DEVICE_VERY_SHORT_WIDTH
  ? csstyles.vars.csBoxSpacing
  : csstyles.vars.csBoxSpacing2x

const styles = StyleSheet.create({
  title: {
    ...csstyles.text.bold,
    ...csstyles.text.textPrimary,
    fontSize: 20,
    paddingLeft: navTitleLeftPadding
  }
})

export default NavTitle
