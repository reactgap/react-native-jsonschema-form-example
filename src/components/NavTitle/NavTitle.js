// @flow
// @format

import React, { PureComponent } from 'react'
import { StyleSheet, Text, type TextStyle, View, Image } from 'react-native'
import csstyles from '../../csstyles'
import { IS_DEVICE_VERY_SHORT_WIDTH } from '../../utils/deviceHelper'

type Props = {
  title?: string,
  hasLogo?: Boolean,
  style?: TextStyle,
  numberOfLines?: number
}

class NavTitle extends PureComponent<Props> {
  render() {
    const { titleCode, t, style, title, numberOfLines, hasLogo } = this.props

    return (
      <View style={styles.container}>
        {
          hasLogo && <Image
          source={{
            uri: 'logo_header'
          }}
          resizeMode={'contain'}
          style={styles.logo}
          />
        }
        <Text style={[styles.title, style]} numberOfLines={numberOfLines}>
          {titleCode ? titleCode : title}
        </Text>
      </View>
    )
  }
}

const navTitleLeftPadding = IS_DEVICE_VERY_SHORT_WIDTH
  ? csstyles.vars.csBoxSpacing
  : csstyles.vars.csBoxSpacing2x

const styles = StyleSheet.create({
  container:{
    flex: 1,
    ...csstyles.base.rowCenter
  },
  logo: {
    width: 62,
    height: 18,
  },
  title: {
    ...csstyles.text.bold,
    ...csstyles.text.textWhite,
    fontSize: 20,
    paddingLeft: navTitleLeftPadding
  }
})

export default NavTitle
