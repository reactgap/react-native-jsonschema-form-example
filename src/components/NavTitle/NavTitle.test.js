import React from 'react'
import { Dimensions, Platform} from 'react-native'
import {shallow} from 'enzyme'
import NavTitle from './NavTitle'
import Adapter from 'enzyme-adapter-react-16'

describe('NavTitle', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<NavTitle title="test label" numberOfLine={1}/>)
            expect(component).toMatchSnapshot()
        });
    });
});