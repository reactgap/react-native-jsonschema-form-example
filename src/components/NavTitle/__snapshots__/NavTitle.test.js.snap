// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTitle Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
